File: answers/q2/scrapper.py
# Install the libraries for web scraping i.e BeautifulSoup, requests, and urllib.
# Make a request to the url
# Get the required content
# Extract the  data

import requests
from bs4 import BeautifulSoup

# make a request from a given url
url = 'https://trial.com'
response = requests.get(url)

# parse content using beautifulsoup
soup = BeautifulSoup(response.content, 'html.parser')

# Get the required dataa
title = soup.title.text  #get the title
paragraph = soup.find_all('p', {'class': 'hello'}) #find all p tags with hthe given class name

print('Title:', title)
print('Paragraph:', paragraph)

File: answers/q3/scrapper.js
// Make sure your environment can run nodejs. Use npm install and follow prompts to set up th node project
// Install the libraries for web scraping i.e cheerio and axios: npm install axios
// Make a request to the url
// Get the required content
// Extract the  data
//The call back function includes a error block to catch errors


const axios = require('axios');
const cheerio = require('cheerio');

// make a request
const url = 'https://trial.com';
axios.get(url)
  .then(response => {
    // parse the HTML content
    const $ = cheerio.load(response.data);

    // extract data
    const title = $('title').text();
    const paragraphs = $('p').map((i, el) => $(el).text()).get();

    console.log('Title:', title);
    console.log('Paragraphs:', paragraphs);
  })
  .catch(error => {
    console.log(error);
  });
